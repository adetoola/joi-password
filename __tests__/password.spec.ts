import Joi from '@hapi/joi';

import passwordValidation from '../src';

const CustomJoi = Joi.extend(passwordValidation);

const schema = CustomJoi.object({
  password: CustomJoi.string().password()
});

describe('Joi Password Validation', () => {
  it('should fail if password is weak', () => {
    const { error } = schema.validate({
      password: 'r'
    });

    expect(error).toMatchSnapshot();
  });

  it('should pass when password is strong', () => {
    const { value } = schema.validate({
      password: 'r@@t !s Passw0r9'
    });

    expect(value).toMatchObject({
      password: 'r@@t !s Passw0r9'
    });
  });
});
