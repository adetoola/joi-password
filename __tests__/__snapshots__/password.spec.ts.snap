// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Joi Password Validation should fail if password is weak 1`] = `[ValidationError: ""password"" must be a strong password]`;
